File: src/main/java/uk/yermak/audiobookconverter/ArtWorkBean.java
package uk.yermak.audiobookconverter;

import java.io.File;

public class ArtWorkBean implements ArtWork {
    private String fileName;
    private long crc32;

    public String fileName() {
        return this.fileName;
    }

    public long crc32() {
        return this.crc32;
    }

    public long getCrc32() {
        return this.crc32();
    }

    public String getFileName() {
        return this.fileName();
    }

    @Override
    public boolean matchCrc32(long crc32) {
        return this.crc32 == crc32;
    }

    public ArtWorkBean(final String fileName, final long crc32) {
        this.fileName = fileName;
        this.crc32 = crc32;
    }

    public ArtWorkBean(String fileName) {
        this(fileName, Utils.checksumCRC32(new File(fileName)));
    }

}

        

File: src/main/scala/uk/yermak/audiobookconverter/ArtWorkBean.scala
package uk.yermak.audiobookconverter

/**
  * Created by yermak on 1/11/2018.
  */
class ArtWorkBean(var fileName: String, var crc32: Long) extends ArtWork {
  override def getCrc32: Long = crc32

  def setCrc32(crc32: Long): Unit = this.crc32 = crc32

  override def getFileName: String = fileName

  override def setFileName(fileName: String): Unit = this.fileName = fileName
}